clearwater-infrastructure/etc/bash.bashrc.clearwater

```diff
@@ -1,22 +1,30 @@
+# Read out the node types on the server from the installed packages
 type=$(. /etc/clearwater/config
        {
+         # List all installed packages for CentOS or Debian. rpm returns an unordered list, dpkg-query is already sorted
          if which yum > /dev/null 2>&1 ; then
-           rpm -qa --qf '%{NAME}\n'
+           rpm -qa --qf '%{NAME}\n' | sort | uniq
          else
            dpkg-query -W -f='${Package}\n'
          fi
        } |
-       egrep '^(bono|clearwater-sip-stress|ellis|homer|homestead|sprout|ralf)$' |
+       egrep '^(bono|clearwater-sip-stress|ellis|homer|homestead|sprout|stats-engine|ralf)$' |
+       # Replacements with customer-friendly names and concatenate with hyphens
        sed -e 's/clearwater-sip-stress/sipp/g' |
+       sed -e 's/stats-engine/mvse/g' |
        tr "\\n" "-" |
-       sort |
        sed -e 's/bono-ellis-homer-homestead-sprout/cw-aio/g
                s/-$//g')
+
+# Read out the node index if set
 node_idx=$(. /etc/clearwater/config
            if [ -n "$node_idx" ]
            then
              echo -$node_idx
            fi)
+
+# Set the format of the bash prompt to be "[NODETYPES-NODEID]USER@HOST:WORKINGDIRECTORY$ "
+# whereas NODETYPES is a hyphenated concatenation of all installed nodes in alphabetical order (e.g. "bono-homestead-sprout-ralf")
```

Comment:
Once again, this explains what the code does rather than why. How about:

`# The node type and index should be present in the command prompt.`